### PR TITLE
r.random.surface: added test file for r.random.surface module

### PR DESCRIPTION
**Pull request by jayneel-shah18**
This PR introduces a comprehensive suite of tests for the `r.random.surface` GRASS GIS module. The test suite is designed to validate core functionality, assess statistical and spatial properties, and handle edge cases to ensure robust module behavior.

## Key Updates in this PR:

- **Functionality:** Validate raster output generation, seed reproducibility, and behavior with masks.  
- **Properties:** Test statistical characteristics (mean, standard deviation, uniformity) and spatial correlation.  
- **Edge Cases:** Verify outputs with varying seeds and custom high-value settings.  
- **Advanced Testing:** Assess uniform distributions and behavior under different distance and exponent parameters.  

### Basic Functionality
- Validate reproducibility with the same seed (`test_seed_reproducibility`).
- Ensure raster values are generated with a mask applied (`test_with_mask`).

### Advanced Properties
- Statistical Properties: Confirm mean and standard deviation fall within expected ranges (`test_distribution_properties`).
- Spatial Correlation: Check spatial dependencies using distance parameters (`test_regional_correlation`).

### Edge Cases
- Different Seeds: Ensure unique outputs for varying seeds (`test_different_seeds`).
- Custom High Value: Verify that generated values do not exceed the specified maximum (`test_custom_high`).

### Advanced Testing
- Uniform Distribution: Ensure values follow a uniform distribution when the `-u` flag is used (`test_uniform_distribution`).

Future enhancements could include performance benchmarking for large-scale regions and stress testing with extreme parameters.

### CODE REVIEW

File: raster/r.random.surface/testsuite/test_r_random_surface.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+    def test_with_mask(self):
+        """Test with a raster mask applied."""
+        self.runModule("r.mask", raster=self.mask_raster)
+        self.assertModule(
+            "r.random.surface", output=self.output_raster, seed=42, overwrite=True
+        )
+        self.runModule("r.mask", flags="r")
+
+        values = gscript.read_command(
```
Comment (petrasovaa) [inline raster/r.random.surface/testsuite/test_r_random_surface.py:70]: I would test specifically expected number of non-null cells with the  univar assert

File: raster/r.random.surface/testsuite/test_r_random_surface.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+        self.runModule(
+            "r.random.surface",
+            output=self.output_raster,
+            distance=5.0,
+            exponent=3.0,
+            seed=42,
+            overwrite=True,
+        )
+        result2 = gscript.read_command(
```
Comment (petrasovaa) [inline raster/r.random.surface/testsuite/test_r_random_surface.py:117]: This would be more useful test if you compared the values to a refernce, basically we need regression testing as we discussed before. Testing that a tool has different values when you change parameters is not super useful.

File: raster/r.random.surface/testsuite/test_r_random_surface.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
+import numpy as np
+from grass.script import core as gscript
```
Comment (petrasovaa) [inline raster/r.random.surface/testsuite/test_r_random_surface.py:2]: `import grass.script as gs`

File: raster/r.random.surface/testsuite/test_r_random_surface.py [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        self.assertNotEqual(
+            result1, result2, "Different distance parameters produced identical outputs"
+        )
+
+    def test_distribution_properties(self):
+        """Test statistical properties of the generated random surface."""
+        self.runModule(
+            "r.random.surface", output=self.output_raster, seed=42, overwrite=True
+        )
+        stats = gscript.parse_command("r.univar", map=self.output_raster, flags="g")
```
Comment (petrasovaa) [inline raster/r.random.surface/testsuite/test_r_random_surface.py:130]: use json output (json fomat is in main branch only, but that's fine)
https://grass.osgeo.org/grass-devel/manuals/r.univar.html

```
gs.parse_command("r.univar", ..., format="json")
```

File: raster/r.random.surface/testsuite/test_r_random_surface.py (hunk 1 above)
Comment (jayneel-shah18) [reply raster/r.random.surface/testsuite/test_r_random_surface.py]: I have updated the test to explicitly check the expected number of non-null cells using `assertRasterFitsUnivar()`

File: raster/r.random.surface/testsuite/test_r_random_surface.py (hunk 2 above)
Comment (jayneel-shah18) [reply raster/r.random.surface/testsuite/test_r_random_surface.py]: I have updated the test to compare the output to reference statistics for the distance parameter, implementing regression testing as discussed.

File: raster/r.random.surface/testsuite/test_r_random_surface.py (hunk 3 above)
Comment (jayneel-shah18) [reply raster/r.random.surface/testsuite/test_r_random_surface.py]: Thanks for pointing that out! I have updated the import to `import grass.script as gs` as requested.

File: raster/r.random.surface/testsuite/test_r_random_surface.py (hunk 4 above)
Comment (jayneel-shah18) [reply raster/r.random.surface/testsuite/test_r_random_surface.py]: I have updated the code to use the json format for `gs.parse_command('r.univar')` as per the documentation.

File: raster/r.random.surface/testsuite/test_r_random_surface.py [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+            output=self.output_raster,
+            seed=42,
+            flags="u",
+            overwrite=True,
+        )
+
+        stats_json = gs.read_command(
+            "r.univar", map=self.output_raster, flags="g", format="json"
+        )
+        stats = json.loads(stats_json)
```
Comment (petrasovaa) [inline raster/r.random.surface/testsuite/test_r_random_surface.py:230]: This can be shortened using the parse_command:
```suggestion
        stats = gs.parse_command("r.univar", map=self.output_raster, format="json")
```

File: raster/r.random.surface/testsuite/test_r_random_surface.py (hunk 4 above)
Comment (petrasovaa) [reply raster/r.random.surface/testsuite/test_r_random_surface.py]: This still needs to be improved, see my other comments

File: raster/r.random.surface/testsuite/test_r_random_surface.py (hunk 5 above)
Comment (jayneel-shah18) [reply raster/r.random.surface/testsuite/test_r_random_surface.py]: Updated the file with the suggested changes and removed float conversions in the following lines.

Comment (petrasovaa) [reply raster/r.random.surface/testsuite/test_r_random_surface.py]: Great, but there are 3 other cases like that.

Comment (jayneel-shah18) [reply raster/r.random.surface/testsuite/test_r_random_surface.py]: I am sorry, I had overlooked the other cases. I have now updated the tests to use the parse command.

**Review by petrasovaa — APPROVED** [review]
